app/models/miq_worker/systemd_common.rb

```diff
@@ -0,0 +1,172 @@
+class MiqWorker
+  module SystemdCommon
+    extend ActiveSupport::Concern
+
+    class_methods do
+      def ensure_systemd_files
+        target_file_path.write(target_file) unless target_file_path.exist?
+        service_file_path.write(unit_file) unless service_file_path.exist?
+      end
+
+      def service_base_name
+        minimal_class_name.underscore.tr("/", "_")
+      end
+
+      def slice_base_name
+        "miq"
+      end
+
+      def service_name
+        scalable? ? service_base_name : "#{service_base_name}@"
+      end
+
+      def service_file_name
+        "#{service_name}.service"
+      end
+
+      def slice_name
+        "#{slice_base_name}-#{service_base_name}.slice"
+      end
+
+      def service_file_path
+        systemd_unit_dir.join(service_file_name)
+      end
+
+      def target_file_name
+        "#{service_base_name}.target"
+      end
+
+      def target_file_path
+        systemd_unit_dir.join(target_file_name)
+      end
+
+      def systemd_unit_dir
+        Pathname.new("/etc/systemd/system")
+      end
+
+      def target_file
+        <<~TARGET_FILE
+          [Unit]
+          PartOf=miq.target
+        TARGET_FILE
+      end
+
+      def unit_file
+        <<~UNIT_FILE
+          [Unit]
+          PartOf=#{target_file_name}
+          [Install]
+          WantedBy=#{target_file_name}
+          [Service]
+          WorkingDirectory=#{working_directory}
+          ExecStart=/bin/bash -lc '#{exec_start}'
+          Restart=always
+          Slice=#{slice_name}
+        UNIT_FILE
+      end
+
+      def working_directory
+        # TODO: pull this dynamically
+        "/var/www/miq/vmdb"
```

Comment:
Is this not `Rails.root`?

Reply:
Yes, I had it hard coded because this doesn't actually work as a non-root user yet...I was hoping to enable the unprivileged user mode and use Rails.root but haven't gotten to it.  That said running as root user isn't exactly equal to your Rails.root == /var/www/miq/vmdb so I'll go back to using Rails.root.